README.md

```diff
@@ -18,53 +14,113 @@ npm i @opengovsg/sgid-client
 
 ## Usage
 
+For more detailed instructions on how to register your client and integrate with sgID, please refer to our [developer documentation](https://docs.id.gov.sg/).
+
 ### Initialization
 
 ```typescript
-import SgidClient from '@opengovsg/sgid-client'
+import { SgidClient } from '@opengovsg/sgid-client'
 
 const client = new SgidClient({
-  clientId: 'CLIENT-ID',
-  clientSecret: 'cLiEnTsEcReT',
-  privateKey: '-----BEGIN PRIVATE KEY-----MII ... XXX-----END PRIVATE KEY-----',
-  redirectUri: 'http://localhost:3000/callback',
+  clientId: '<Insert your client ID>',
+  clientSecret: '<Insert your client secret>',
+  privateKey: '<Insert your private key as a string>',
+  redirectUri: '<Insert your redirect URI>',
 })
 ```
 
+### Generate code challenge and verifier
+
+This is a required step for OIDC ([more info](https://oauth.net/2/pkce/)).
+
+- The code challenge should be provided in `authorizationUrl`.
+- The codeVerifier should be stored in the user's session so it can be retrieved later for use in `callback`.
+- A unique pair should be generated for each authorization request.
+
+`generatePkcePair([length])`
+
+- parameters: `<number>` (Optional) Length of the codeVerifier to generate. **Default**: 43
+- Returns: `<Object>`
+  - codeChallenge: `<string>` **S256** code challenge generated from the code verifier
+  - codeVerifier: `<string>`
+
+Example usage:
+
+```typescript
+const { codeChallenge, codeVerifier } = generatePkcePair()
+```
+
 ### Get Authorization URL
 
-`client.authorizationUrl(state, scope, [nonce], [redirectUri])`
+`client.authorizationUrl(parameters)`
+
+- parameters: `<Object>`
+  - codeChallenge: `<string>` Code challenge generated from the code verifier.
+  - state: `<string>` (Optional) Used to track application state which will be passed back to you in query params via the redirect URI after login.
+  - scope: `<string> | <string[]>` (Optional) Scopes being requested. Can be provided as a string array or a space-concatenated string. **Default**: `myinfo.name openid`
+  - nonce: `<string> | <null>` (Optional) Random, unique value to associate a user-session with an ID Token and to mitigate replay attacks. Set as `null` to omit the nonce. **Default**: Randomly generates a nonce if unspecified or set as `undefined`
+  - redirectUri: `<string>` (Optional) Overrides redirect URI initially provided in constructor. **Default**: Utilizes the `redirectUri` provided in the constructor.
+- Returns: `<Object>`
+  - url: `<string>` Generated authorization url.
+  - nonce: `<string> | <undefined>` Provided nonce, randomly generated nonce, or `undefined` (based on nonce input).
```

Comment:
just as with code verifier, we can let them know that the nonce should be stored in the user's session so it can be retrieved in `callback` later

Reply:
makes sense!